testnet/stacks-node/src/tests/signer/commands/commit_ops.rs

```diff
@@ -0,0 +1,89 @@
+use super::context::{SignerTestContext, SignerTestState};
+use madhouse::{Command, CommandWrapper};
+use proptest::prelude::{Just, Strategy};
+use stacks::util::tests::TestFlag;
+use std::sync::Arc;
+
+pub struct SkipCommitOpPrimaryMiner {
+    miner_1_skip_commit_flag: TestFlag<bool>,
+}
+
+impl SkipCommitOpPrimaryMiner {
+    pub fn new(miner_1_skip_commit_flag: TestFlag<bool>) -> Self {
+        Self {
+            miner_1_skip_commit_flag,
+        }
+    }
+}
+
+impl Command<SignerTestState, SignerTestContext> for SkipCommitOpPrimaryMiner {
+    fn check(&self, state: &SignerTestState) -> bool {
+        println!(
+            "Checking: Skipping commit operations for miner 1. Result: {:?}",
+            !state.is_primary_miner_skip_commit_op
+        );
+        !state.is_primary_miner_skip_commit_op
+    }
+
+    fn apply(&self, state: &mut SignerTestState) {
+        println!("Applying: Skipping commit operations for miner 1");
+
+        self.miner_1_skip_commit_flag.set(true);
+
+        state.is_primary_miner_skip_commit_op = true;
+    }
+
+    fn label(&self) -> String {
+        "SKIP_COMMIT_OP_PRIMARY_MINER".to_string()
+    }
+
+    fn build(
+        ctx: Arc<SignerTestContext>,
```

Comment:
I don't think the trait should prescribe an `Arc` here; `TestContext` is already `Clone`. If the implementer needs `Send + Sync` then they should internally `Arc` the necessary fields imo.